gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixJobScheduler.java

```diff
@@ -324,10 +330,17 @@ public void handleUpdateJobConfigArrival(UpdateJobConfigArrivalEvent updateJobAr
   }
 
   @Subscribe
-  public void handleDeleteJobConfigArrival(DeleteJobConfigArrivalEvent deleteJobArrival) {
+  public void handleDeleteJobConfigArrival(DeleteJobConfigArrivalEvent deleteJobArrival) throws InterruptedException {
     LOGGER.info("Received delete for job configuration of job " + deleteJobArrival.getJobName());
     try {
       unscheduleJob(deleteJobArrival.getJobName());
+      Properties jobConfig = deleteJobArrival.getJobConfig();
+      if (PropertiesUtils.getPropAsBoolean(jobConfig, GobblinClusterConfigurationKeys.SHOULD_CANCEL_RUNNING_JOB_ON_DELETE, "false")) {
```

Comment:
Should the default be false or true? 
It could be argued that the expected behavior on a job spec deletion is that running jobs should be cancelled? 

Are you just trying to maintain backward compatibility of this with current jobs?

Reply:
I think it has to be `False` since to update the job config `handleUpdateJobConfigArrival` method deletes and insert as new job config. Or if we want it to be `True` then the method `handleUpdateJobConfigArrival` has to override this config while updating the jobConfig. #my2Cents.

Reply:
Yes, it is to maintain backward compatibility.

Reply:
@jhsenjaliya +1 for noticing that. If the default were true, it will impact the job config updates as well.